tests/saved-state.test.js

```diff
@@ -1,37 +1,51 @@
-import { exec } from "child_process";
+import { execSync } from "child_process";
 import fs from "fs";
 import path from "path";
 import yaml from "js-yaml";
 import Redis from "ioredis";
 
-function waitForProcess() {
-  let callback = null;
-  const p = new Promise((resolve) => {
-    callback = (/*error, stdout, stderr*/) => {
-      //console.log(stdout);
-      resolve(0);
-    };
-  });
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
 
-  return { p, callback };
+async function waitContainer(containerId) {
+  try {
+    execSync(`docker kill -s SIGINT ${containerId}`);
+  } catch (e) {
+    return;
+  }
+
+  // containerId is initially the full id, but docker ps
+  // only prints the short id (first 12 characters)
+  containerId = containerId.slice(0, 12);
+
+  while (true) {
+    try {
+      const res = execSync("docker ps -q", { encoding: "utf-8" });
+      if (res.indexOf(containerId) < 0) {
+        return;
+      }
+    } catch (e) {
+      console.error(e);
+    }
+    await sleep(500);
+  }
 }
 
-var savedStateFile;
-var state;
-var numDone;
-var redis;
-var finishProcess;
+let savedStateFile;
+let state;
+let numDone;
+let numQueued;
+let finished;
 
 test("check crawl interrupted + saved state written", async () => {
-  let proc = null;
-
-  const wait = waitForProcess();
+  let containerId = null;
 
   try {
-    proc = exec(
-      "docker run -v $PWD/test-crawls:/crawls -v $PWD/tests/fixtures:/tests/fixtures webrecorder/browsertrix-crawler crawl --collection int-state-test --url https://webrecorder.net/ --limit 20",
-      { shell: "/bin/bash" },
-      wait.callback,
+    containerId = execSync(
+      "docker run -d -v $PWD/test-crawls:/crawls -v $PWD/tests/fixtures:/tests/fixtures webrecorder/browsertrix-crawler crawl --collection int-state-test --url https://iana.org/ --limit 10",
```

Comment:
Test looks good but does go back to relying on externally-hosted sites, which could be a problem down the road?

Reply:
I took another look, and we actually have a redirect on https://www.webrecorder.net/ -> https://webrecorder.net/ so can switch to that!

Reply:
Updated!